classes/services/PKPStatsService.inc.php

```diff
@@ -0,0 +1,567 @@
+	}
+
+	/**
+	 * Get statistics records of the submissions (filtered by request parameters),
+	 * containing given time segments, abstractViews and totalFileViews,
+	 * ordered DESC by the time segments.
+	 *
+	 * @param int $contextId
+	 * @param array $args {
+	 * 		@option string orderBy
+	 * 		@option string orderDirection
+	 * 		@option int count
+	 * 		@option int offset
+	 *		@option string timeSegment
+	 *		@option string dateStart
+	 *		@option string dateEnd
+	 *		@option array sectionIds
+	 *		@option array searchPhrase
+	 * }
+	 *
+	 * @return array
+	 */
+	public function getTotalSubmissionsStats($contextId, $args = array()) {
+		$records = array();
+		$statsListQB = $this->_buildGetTotalSubmissionsStatsQueryObject($contextId, $args);
+		$statsQO = $statsListQB->get();
+		/* default monthly: SELECT month, assoc_type, SUM(metric) AS metric FROM metrics WHERE context_id = ? AND assoc_type IN (1048585, 515) AND metric_type = 'ojs::counter' GROUP BY month, assoc_type ORDER BY month DESC */
+		/* with filters monthly: SELECT month, assoc_type, SUM(metric) AS metric FROM metrics WHERE context_id = ? AND assoc_type IN (1048585, 515) AND metric_type = 'ojs::counter' AND pkp_section_id = ? AND day BETWEEN ? AND ? AND submission_id = ? GROUP BY month, assoc_type ORDER BY month DESC */
+
+		if ($statsQO) {
+			$dao = \DAORegistry::getDAO('MetricsDAO');
+			$result = $dao->retrieve($statsQO->toSql(), $statsQO->getBindings());
+			$records = $result->GetAll();
+		}
+		return $records;
+	}
+
+	/**
+	 * @see \PKP\Services\EntityProperties\EntityPropertyInterface::getProperties()
+	 *
+	 * @param $entity Submission
+	 * @param $props array
+	 * @param $args array
+	 *		$args['request'] PKPRequest Required
+	 *		$args['slimRequest'] SlimRequest
+	 *		$args['params] array of validated request parameters
+	 *
+	 * @return array
+	 */
+	public function getProperties($entity, $props, $args = null) {
+		$entityService = null;
+		if (is_a($entity, 'Submission')) {
+			$entityService = \ServicesContainer::instance()->get('submission');
+			$params = array(
+				'entityAssocType' => ASSOC_TYPE_SUBMISSION,
+				'fileAssocType' => ASSOC_TYPE_SUBMISSION_FILE,
+			);
+			$statsListQB = $this->_buildGetSubmissionQueryObject($entity->getContextId(), $entity->getId(), $args['params']);
+			$statsQO = $statsListQB->get();
+			/* default monthly: SELECT month, assoc_type, file_type, SUM(metric) AS metric FROM metrics WHERE submission_id = ? AND assoc_type IN (1048585, 515) AND metric_type = 'ojs::counter' GROUP BY month, assoc_type, file_type, month */
+			/* with filters monthly: SELECT month, assoc_type, file_type, SUM(metric) AS metric FROM metrics WHERE submission_id = ? AND assoc_type IN (1048585, 515) AND metric_type = 'ojs::counter' AND day BETWEEN ? AND ? GROUP BY month, assoc_type, file_type, month */
+		}
+
+		$dao = \DAORegistry::getDAO('MetricsDAO');
+		$result = $dao->retrieve($statsQO->toSql(), $statsQO->getBindings());
+		$records = $result->GetAll();
```

Comment:
This is one comment left I still have to see/think about...

Reply:
While hunting down the performance stuff for antti-jussi's database, I came to understand this better, and why there are two sets of database requests. 

I'm comfortable with this being left as-is for now, since the queries don't seem to represent a major performance issue after all.

The metrics are very different kinds of objects, and our existing Service class structure (get -> query-builder -> properties) is not intuitive. You've done well to make it happen and sorry if I have pushed the structure on you without understanding the unique problems that well!